proxyclient/chainload.py

```diff
@@ -31,6 +32,23 @@
     # Fall back to proxy heap, which will be at the right place in old versions
     new_base = u.memalign(0x10000, memory_size)
 
+def compressed_writemem(iface, dest, data, progress):
+    if not len(data):
+        return
+
+    payload = lzma.compress(data)
+    compressed_size = len(payload)
+
+    try:
+        compressed_addr = p.malloc(compressed_size)
```

Comment:
Just always use u.malloc; p.malloc is only useful when the data needs to outlive the execution of the script. I expect all python-side allocations to use the python-side allocator by default (it's faster too).

Reply:
I just blindly copied the code from above without understanding what was going on ;)
Fixed as well.